src/theme/css/general.css

```diff
@@ -14,6 +14,10 @@ html {
     -webkit-text-size-adjust: none;
 }
 
+* {
+    color: var(--fg);
+}
```

Comment:
Okay, so, why is this needed?

Reply:
Without this change, it looks like this:

![image](https://github.com/user-attachments/assets/83010b77-8a8f-42b8-9a7c-4b18b9ea22de)

Reply:
Yes, but why is it happening? This change seems like a workaround more than an actual fix.

Reply:
I can't figure out why. In the inspector, it says that this is the color in use, but for reasons I can't understand it's not being applied.

Reply:
Here's a patch that fixes the problem in both Firefox and Ungoogled Chromium when I try it:

```diff
diff --git a/src/theme/css/chrome.css b/src/theme/css/chrome.css
index 9dbfe2c18..369d59ae0 100644
--- a/src/theme/css/chrome.css
+++ b/src/theme/css/chrome.css
@@ -91,7 +91,7 @@ body.sidebar-visible #menu-bar {
     display: flex;
     margin: 0 5px;
 }
-body:not(.js) .left-buttons button {
+html:not(.js) .left-buttons button {
     display: none;
 }
 
@@ -427,7 +427,7 @@ ul#searchresults span.teaser em {
     -ms-user-select: none;
     user-select: none;
 }
-body:not(.sidebar-resizing) .sidebar {
+html:not(.sidebar-resizing) .sidebar {
     transition: transform 0.3s; /* Animation: slide away */
 }
 .sidebar code {
diff --git a/src/theme/css/general.css b/src/theme/css/general.css
index 0e7443cc7..7670b087d 100644
--- a/src/theme/css/general.css
+++ b/src/theme/css/general.css
@@ -14,10 +14,6 @@ html {
     -webkit-text-size-adjust: none;
 }
 
-* {
-    color: var(--fg);
-}
-
 body {
     margin: 0;
     font-size: 1.6rem;
diff --git a/src/theme/css/variables.css b/src/theme/css/variables.css
index a30b97eba..12d1db7a3 100644
--- a/src/theme/css/variables.css
+++ b/src/theme/css/variables.css
@@ -112,7 +112,7 @@
     --copy-button-filter-hover: invert(36%) sepia(70%) saturate(503%) hue-rotate(167deg) brightness(98%) contrast(89%);
 }
 
-.light, body:not(.js) {
+.light, html:not(.js) {
     --bg: hsl(0, 0%, 100%);
     --fg: hsl(0, 0%, 0%);
 
@@ -258,7 +258,7 @@
 }
 
 @media (prefers-color-scheme: dark) {
-    body:not(.js) {
+    html:not(.js) {
         --bg: hsl(200, 7%, 8%);
         --fg: #98a3ad;
 
diff --git a/src/theme/index.hbs b/src/theme/index.hbs
index ea19458ec..99d304fc1 100644
--- a/src/theme/index.hbs
+++ b/src/theme/index.hbs
@@ -85,7 +85,7 @@
             var html = document.querySelector('html');
             html.classList.remove('{{ default_theme }}')
             html.classList.add(theme);
-            document.body.classList.add("js");
+            html.classList.add("js");
         </script>
 
         <input type="checkbox" id="sidebar-toggle-anchor" class="hidden">
```

Reply:
Wow. This is really impressive. Thanks a lot!